src/taco-cli/test/kit.ts

```diff
@@ -89,19 +114,96 @@ describe("Kit", function (): void {
+        return platform.run({
+            options: {},
+            original: args,
+            remain: args
+        });
+    }
+
+    function pluginRun(args: string[]): Q.Promise<any> {
+        var plugin: ICommand = CommandHelper.getCommand("plugin");
+        return plugin.run({
+            options: {},
+            original: args,
+            remain: args
+        });
+    }
+
+    function addPlatformToProject(platfromName: string, projectPath: string): Q.Promise<any> {
+        process.chdir(projectPath);
+        return platformRun(["add", platfromName]);
+    }
+
+    function addTestPluginsToProject(projectPath: string): Q.Promise<any> {
+        process.chdir(projectPath);
+        return pluginRun(["add", "cordova-plugin-camera"])
+        .then(function(): Q.Promise<any> {
+            return sleep(100);
+        }).then(function (): Q.Promise<any> {
+            var medicPluginTestsPath = path.resolve(".", "plugins", "cordova-plugin-camera", "tests");
+            var pluginCommandArgs: string[] = ["add", medicPluginTestsPath];
+            return pluginRun(pluginCommandArgs);
+        });
+    }
+
+    function getMockYesOrNoHandler(errorHandler: (err: Error) => void, onClose: () => void, desiredResponse: string): {
+        question: (question: string, callback: (answer: string) => void) => void;
+        close: () => void;
+    } {
+        return {
+            question: function (question: string, callback: (answer: string) => void): void {
+                switch (question) {
+                    case "CommandKitSelectProjectUpdatePrompt":
+                        callback(desiredResponse);
+                        break;
+                    default:
+                        errorHandler(new Error("Unexpected query!"));
+                }
+            },
+           close: onClose
+        };
+    }
+
+    function sleep(milliseconds: number): Q.Promise<any> {
+        var deferred: Q.Deferred<any> = Q.defer();
+        setTimeout(deferred.resolve, milliseconds);
+        return deferred.promise;
+    }
+
+    function runKitCommandSuccessCaseAndVerifyTacoJson(args: string[],
         tacoJsonPath: string, tacoJsonKeyValues: IKeyValuePair<string>): Q.Promise<TacoUtility.ICommandTelemetryProperties> {
         return kitRun(args)
         .then((telemetryParameters: TacoUtility.ICommandTelemetryProperties) => {
             fs.existsSync(tacoJsonPath).should.be.true;
 
-            var tacoJson: IKeyValuePair<string> = require(tacoJsonPath);
-
+            var tacoJson: IKeyValuePair<string> = JSON.parse(fs.readFileSync(tacoJsonPath, { encoding: "utf-8" }));
             tacoJsonKeyValues.should.be.eql(tacoJson);
             return telemetryParameters;
         });
     }
 
+    function runKitCommandFailureCaseAndVerifyTacoJson<T>(args: string[],
+        tacoJsonPath: string, tacoJsonKeyValues: IKeyValuePair<string>, expectedErrorCode: T): Q.Promise<TacoUtility.ICommandTelemetryProperties> {
```

Comment:
This function doesn't really return a Q.Promise<TacoUtility.ICommandTelemetryProperties> since it is only the error case. Perhaps Q.Promise<void> ? Then the Q.resolve(null) can be omitted and the final .then handlers can just return void